hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/RestCallCollector.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.ascii.rest;
+
+import com.hazelcast.internal.util.counters.MwCounter;
+
+import java.util.Comparator;
+import java.util.concurrent.ConcurrentSkipListSet;
+
+public class RestCallCollector {
+
+    public static final int SET_SIZE_LIMIT = 1000;
+
+    private static class RequestIdentifier implements Comparable<RequestIdentifier> {
+
+        public static final Comparator<RequestIdentifier> COMPARATOR = Comparator.comparing(RequestIdentifier::getMethod)
+                .thenComparing(RequestIdentifier::getPath);
+
+        private final String method;
+        private final String path;
+
+        RequestIdentifier(String method, String path) {
+            this.method = method;
+            this.path = path;
+        }
+
+        @Override
+        public int compareTo(RestCallCollector.RequestIdentifier o) {
+            return COMPARATOR.compare(this, o);
+        }
+
+        public String getMethod() {
+            return method;
+        }
+
+        public String getPath() {
+            return path;
+        }
+    }
+
+    private final MwCounter mapPostSuccCount = MwCounter.newMwCounter();
+    private final MwCounter mapPostFailCount = MwCounter.newMwCounter();
+    private final MwCounter mapGetSuccCount = MwCounter.newMwCounter();
+    private final MwCounter mapGetFailCount = MwCounter.newMwCounter();
+    private final MwCounter mapDeleteSuccCount = MwCounter.newMwCounter();
+    private final MwCounter mapDeleteFailCount = MwCounter.newMwCounter();
+    private final MwCounter mapTotalRequestCount = MwCounter.newMwCounter();
+    private final MwCounter queuePostSuccCount = MwCounter.newMwCounter();
+    private final MwCounter queuePostFailCount = MwCounter.newMwCounter();
+    private final MwCounter queueGetSuccCount = MwCounter.newMwCounter();
+    private final MwCounter queueGetFailCount = MwCounter.newMwCounter();
+    private final MwCounter queueDeleteSuccCount = MwCounter.newMwCounter();
+    private final MwCounter queueDeleteFailCount = MwCounter.newMwCounter();
+    private final MwCounter queueTotalRequestCount = MwCounter.newMwCounter();
+
+    private final MwCounter requestCount = MwCounter.newMwCounter();
+    private final ConcurrentSkipListSet<RequestIdentifier> uniqueRequests = new ConcurrentSkipListSet<>();
```

Comment:
do we need to have these sets sorted?
If not please consider `ConcurrentHashMap.newKeySet()`